options.go

```diff
@@ -400,43 +389,98 @@ func WithCGroup(useCGroup bool) Option {
 	})
 }
 
-func WithLoggerLevel(level int) Option {
+func WithLoggerLevel(level Lever) Option {
 	return optionFunc(func(opts *options) (err error) {
 		opts.LogLevel = level
 		return
 	})
 }
 
-type loggerOptions struct {
-	RotateEnable    bool
-	SplitLoggerSize int64 // SplitLoggerSize The size of the log split
+func WithLogger(logger Logger) Option {
+	return optionFunc(func(opts *options) (err error) {
+		opts.loggerLock.Lock()
+		defer opts.loggerLock.Unlock()
+		oldLogger := opts.Logger
+		if oldLogger != nil {
+			switch lg := oldLogger.(type) {
+			case *fileLog:
+				old := lg.logger.Load()
+				if old != nil {
+					oldFd, ok := old.(*os.File)
+					if !ok {
+						//nolint
+						fmt.Println("assert fault")
+						return
+					}
+					_ = oldFd.Close()
+				}
+
+			}
+		}
+		opts.Logger = logger
+		return
+	})
 }
 
-func newLoggerOptions() *loggerOptions {
-	return &loggerOptions{
-		RotateEnable:    true,
-		SplitLoggerSize: defaultShardLoggerSize,
+// NewFileLog init logger
+// shardLoggerSize eg. "b/B", "k/K" "kb/Kb" "mb/Mb", "gb/Gb" "tb/Tb" "pb/Pb".
+func NewFileLog(dumpPath string, rotateEnable bool, shardLoggerSize string, loginfo ...string) Logger {
+	f := &fileLog{
+		rotateEnable:            rotateEnable,
+		splitLoggerSizeToString: shardLoggerSize,
+		logger:                  atomic.Value{},
 	}
-}
 
-// WithLoggerSplit set the split log options.
-// eg. "b/B", "k/K" "kb/Kb" "mb/Mb", "gb/Gb" "tb/Tb" "pb/Pb".
-func WithLoggerSplit(enable bool, shardLoggerSize string) Option {
-	return optionFunc(func(opts *options) (err error) {
-		opts.logOpts.RotateEnable = enable
-		if !enable {
+	if rotateEnable {
+		parseShardLoggerSize, err := units.FromHumanSize(shardLoggerSize)
+		if err != nil || (err == nil && parseShardLoggerSize <= 0) {
+			f.splitLoggerSize = defaultShardLoggerSize
+		} else {
+			f.splitLoggerSize = parseShardLoggerSize
+		}
+	}
+
+	filePath := path.Join(dumpPath, defaultLoggerName)
+	if len(loginfo) > 0 {
+		filePath = dumpPath + "/" + path.Join(loginfo...)
+	}
+
+	f.dumpPath = filepath.Dir(filePath)
+
+	var (
+		logger *os.File
+		err    error
+	)
+
+	logger, err = os.OpenFile(filepath.Clean(filePath), defaultLoggerFlags, defaultLoggerPerm)
```

Comment:
logger may not be a good name now since there is a Logger interface now.
file may be better?